### ISSUE TITLE
Ability to report undesirable comments

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#2694]
## User story 
- **As an Open Collective user**, I can report comments that doesn't respect the standards of my community or that are illegal in my country (abusive content, racism...etc)
- **As a platform**, we are legally obligated to moderate the content posted on the website. For example the [French law](https://www.theguardian.com/world/2019/jul/09/france-online-hate-speech-law-social-media) now requires you to delete "obviously hateful" content in less than 24h hours if you don't want to be fined. They only way for us to tackle this is to have a reporting system.

## Best solution for this problem
- Whenever users find an inappropriate comment, they must be able to report it
- We need to define clear actions taken #2650

Because expenses are meant to be bundled in conversations with #2394, we should base this report system on Conversations so that we'll be able to use the same system for expenses.

## MVP

Working on a debate platform I had developed a `Report` button that triggered this popup:

![image](https://user-images.githubusercontent.com/1556356/70310978-0d9b1780-1811-11ea-844d-961aca59ab5e.png)

I'd like to implement the same thing, using our own design system and a different categorization:

- Spamming
- Sexualized content
- Threatening or harassing other users
- Promotes violent behaviors
- Discrimination
- Invasion of privacy

## Metrics 
- We should track the number of reports and their types in Forest
- We should ideally have a Slack channel where new reports get posted

## Documentation 
We should clearly document what we consider as a abusive content and what are the repercussions based on the abuse type (example: harassment = direct ban, 3 x SPAM = limited account...etc). Should be part of #2650.

## Related

- Issues about spammers: #2650
- [A good (and short) article about how to handle moderation on social platforms](https://www.getopensocial.com/blog/community-management/manage-abusive-content)
- [A quick look](https://www.coe.int/en/web/no-hate-campaign/reporting-on-social-media-platforms) at what giant platforms are doing

**Comment by alanna** [comment]
Thanks for bringing this up @Betree - we definitely need a solution here.

I think this system should be for not only comments but also expenses and contributions, because those are a big spam issue for us.

My suggestion for the reasons list would be:
* Spam or unwanted commercial content
* Abusive or inappropriate language
* Irrelevant or erroneous content

I think that should cover everything. Maybe we should also have an "other" option and if users pick that often we can look into what's missing on the list.

Some questions:
* Will these reports go to platform admins, Host admins, or Collective admins? Who will be expected to take action? 
* What options will moderators have once they receive a report? How about they can choose one or more of these options....
  * Reply
  * Remove content
  * Block user 
* If we offer blocking, what effect does it have? If it's at the Collective level, it makes sense that the user would be blocked from interacting with the Collective, including comments, making contributions, and submitting expenses. 
* If these reports are going to platform admins or host admins, would that apply to just the Collective or the whole host or whole platform?
* If you remove content and/or block and the content was a financial contribution, does the system refund them and remove the contributor from the Collective's page entirely? I think this is what many Collective admins would want, but we should be clear about the impact.

**Comment by Betree** [comment]
> Will these reports go to platform admins, Host admins, or Collective admins? Who will be expected to take action?

Platform admins would be the easiest to start, tho a page for host/collective admins to manage reports and take action would be a great iteration.

> If we offer blocking, what effect does it have? If it's at the Collective level, it makes sense that the user would be blocked from interacting with the Collective, including comments, making contributions, and submitting expenses.

That would be the best option, that's what Github does (block at the repo or organization level). 

The low hanging fruit is #2695 and adding the ability to ban user at the platform level. Not perfect, but it will give us the ability to block users that are obviously trolls/spammers (that we usually want to prevent from interacting with all collective) at the platform level.

**Comment by alanna** [comment]
@Betree are we still planning to create a flag/report option? I just rejected a spam expense and thought it would be great if admins could report it and we had a process for blocking those users, instead of having to report them to you personally.

**Comment by Betree** [comment]
Definitely, to me this is one of the most important steps to fight spam. This MVP for comments should bring a generic modal that we'll be able to re-use to report expenses or collectives.

**Comment by alanna** [comment]
I still want this! We should have report options for comments and expenses.

**Comment by Betree** [comment]
Added `Tries to impersonate another profile` to the report reasons, as we've been confronted with that a couple of times.

**Comment by alanna** [comment]
As we are tackling platform ethics policy stuff now, this will be important to build.

**Comment by Betree** [comment]
This is kinda supported by the "Mark as SPAM" button for expenses, you can't specify a reason but that's already super helpful.

Would be great to have the same for comments, we've seen a lot of SPAM in there lately.

**Comment by RCheesley** [comment]
Yes indeed @Betree - it's one of the channels we're thinking probably needs to come into our risk assessment, particularly comments on updates or in conversations.
